src/bindings/python/flux/util.py

```diff
@@ -419,6 +419,9 @@ def format_field(self, value, spec):
             basecases = ("", "0s", "0.0", "0:00:00", "1970-01-01T00:00:00")
             value = "-" if str(value) in basecases else str(value)
             spec = spec[:-1] + "s"
+        elif spec.endswith("i"):
```

Comment:
After seeing this duplicated code it took me a second to remember that we can't just handle `inf` via `"h"` because `i` is needed to specifically handle limits which may have a valid and presentable value of `0`. To remind us of this I'd suggest a comment here describing why `i` is being added.

Also, the code could be deduplicated a bit by handling both `h` and `i` in the same code block, e.g. something like
```diff
diff --git a/src/bindings/python/flux/util.py b/src/bindings/python/flux/util.py
index deb13746f..2cb42f42b 100644
--- a/src/bindings/python/flux/util.py
+++ b/src/bindings/python/flux/util.py
@@ -338,6 +338,13 @@ def fsd(secs):
 class UtilFormatter(Formatter):
     # pylint: disable=too-many-branches
 
+    # supported presentation characters which turn a base case
+    # value, e.g. '0' or an empty string, into a single hyphen.
+    hyphenators = {
+        "h": ("", "0s", "0.0", "0:00:00", "1970-01-01T00:00:00"),
+        "i": ("inf"),
+    }
+
     def convert_field(self, value, conv):
         """
         Flux utility-specific field conversions. Avoids the need
@@ -410,8 +417,8 @@ class UtilFormatter(Formatter):
             denote_truncation = True
             spec = spec[:-1]
 
-        if spec.endswith("h"):
-            basecases = ("", "0s", "0.0", "0:00:00", "1970-01-01T00:00:00")
+        if spec and spec[-1] in self.hyphenators:
+            basecases = self.hyphenators[spec[-1]]
             value = "-" if str(value) in basecases else str(value)
             spec = spec[:-1] + "s"
         retval = super().format_field(value, spec)
```